### PR TITLE
Added chapter about sitemap-provider and dump-sitemap command

### PR DESCRIPTION
**Pull request by wachterjohannes**
| Q | A |
| --- | --- |
| Fixed tickets | none |
| License | MIT |
#### What's in this PR?

This PR add the chapter about sitemap provider and dump command.

### CODE REVIEW

File: cookbook/dump-sitemap.rst [hunk 1]
```diff
@@ -0,0 +1,14 @@
+Improve Sitemap Speed
+=====================
+
+The sitemap of sulu is based on small pices which are generated by
```
Comment (danrot) [inline cookbook/dump-sitemap.rst]: `Sulu` and `pieces, wich are generated`

File: cookbook/dump-sitemap.rst [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+so-called `SitemapProvider` (see :doc:`sitemap-provider`). Each
```
Comment (danrot) [inline cookbook/dump-sitemap.rst]: `so called`

File: cookbook/dump-sitemap.rst [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+provider returns mostly 50000 links which can take a bigger amount
```
Comment (danrot) [inline cookbook/dump-sitemap.rst]: `can return many links, which would take a bigger amount of time.`

File: cookbook/dump-sitemap.rst [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+of time. The Google bot is quite busy and dont want to wait long
```
Comment (danrot) [inline cookbook/dump-sitemap.rst]: `The Google does not wait a long time for the sitemap to be returned.

File: cookbook/dump-sitemap.rst [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+for the sitemap to load - therefor Sulu is able to pre-generate the
```
Comment (danrot) [inline cookbook/dump-sitemap.rst]: `therefore`

File: cookbook/dump-sitemap.rst [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+whole sitemap and cache it on the filesystem. This can be triggered
+by calling the following command by a cron-job each night.
```
Comment (danrot) [inline cookbook/dump-sitemap.rst]: `by calling the following command. This can also be done by a cron-job.`

Comment (danrot) [inline cookbook/dump-sitemap.rst]: Otherwise it reads as it can only be done by cronjobs.

File: cookbook/sitemap-provider.rst [hunk 7]
```diff
@@ -0,0 +1,88 @@
+Provider for XML-Sitemap
+========================
+
+`SitemapProvider` are used to load data for the XML-Sitemap. It returns
+an array of `SitemapUrl` instances. This Api has to be paginated because
+Google only allows `SitemapProviderInterface::PAGE_SIZE = 50000` urls in
```
Comment (danrot) [inline cookbook/sitemap-provider.rst]: Wouldn't write the constant in here like that. maybe `Google only allows 50000 urls`? If you think it is necessary I would mention the constant in a separate sentence.

**Comment by wachterjohannes** [comment]
@danrot fixed

**Pull request merged**